### PR TITLE
IRGen: Co-locate metadata instatiation/completions/accessor functions in a special section

### PR DESCRIPTION
**Pull request by aschwaighofer**
For spatial locality on startup.

rdar://101593202

### CODE REVIEW

**Comment by aschwaighofer** [comment]
@swift-ci smoke test

File: lib/IRGen/GenDecl.cpp [hunk 1]
```diff
@@ -5850,3 +5852,24 @@ IRGenModule::getOrCreateHelperFunction(StringRef fnName, llvm::Type *resultTy,
...
+  case llvm::Triple::SPIRV:
+  case llvm::Triple::UnknownObjectFormat:
+    break;
+  case llvm::Triple::Wasm:
+  case llvm::Triple::ELF:
+    f->setSection("swift_colocate_functions");
+    break;
+  case llvm::Triple::XCOFF:
+  case llvm::Triple::COFF:
+    f->setSection(".sw5colo$B");
```
Comment (compnerd) [inline lib/IRGen/GenDecl.cpp:5872]: Do we intend to need to create a linker table for this data?  The grouping (`$B`) is for cases where we need to order the data in between start/stop markers to allow the table computation.  If so, we should probably have an associated set of changes to `SwiftRT-ELF.cpp` and `SwiftRT-COFF.cpp`.

Comment (aschwaighofer) [reply lib/IRGen/GenDecl.cpp:5872]: No. I just copied the string from further up the file.

Comment (compnerd) [reply lib/IRGen/GenDecl.cpp:5872]: In that case, this should probably be named `.sw5colo`.

Comment (aschwaighofer) [reply lib/IRGen/GenDecl.cpp:5872]: 👍 will do

**Comment by aschwaighofer** [comment]
@swift-ci test

**Comment by aschwaighofer** [comment]
@swift-ci test

**Comment by aschwaighofer** [comment]
@swift-ci test source compat

**Comment by aschwaighofer** [comment]
@swift-ci test source compat

**Comment by aschwaighofer** [comment]
@swift-ci test source compat release

**Pull request merged**